rasa/core/processor.py

```diff
@@ -636,12 +634,24 @@ async def _cancel_reminders(
                         scheduler.remove_job(scheduled_job.id)
 
     async def _run_action(
-        self, action, tracker, output_channel, nlg, policy=None, confidence=None
+        self,
+        action,
+        tracker,
+        output_channel,
+        nlg,
+        policy=None,
+        confidence=None,
+        metadata: Optional[Dict] = None,
     ) -> bool:
         # events and return values are used to update
         # the tracker state after an action has been taken
         try:
-            events = await action.run(output_channel, nlg, tracker, self.domain)
+            if action.name() == ACTION_SESSION_START_NAME:
+                events = await action.run(
+                    output_channel, nlg, tracker, self.domain, metadata=metadata
```

Comment:
if we keep it in the other actions' signature, we should emit a `UserWarning` indicating that the argument will have no effect

Reply:
@ricwo since all other actions override Action::run(), they have to comply with the signature right? One solution to avoid the `UserWarning` is to add `**kwargs` instead of explicit `metadata=None` for those actions that don't need it.

Reply:
The other option I can think of is to pass metadata to the constructor of `ActionSessionStart`, but that'd be quite a refactroing.

Reply:
> The other option I can think of is to pass metadata to the constructor of ActionSessionStart, but that'd be quite a refactroing.

I like that idea. instead of overriding the `__init__()`, we could add a class (see screenshot) and after the `if action.name() == ACTION_SESSION_START_NAME:` in l. 649 in processor.py, set `action.metadata = metadata` before calling `action.run`. 

![image](https://user-images.githubusercontent.com/20581300/83865241-394a7c00-a726-11ea-8473-15ad66df8934.png)

Reply:
That's also an option 👍 
I thought about it as well, but that looked a little bit hacky to me tbh.
But still, I think I like it more than adding metadata to the parent's class run().